PR to support postgres-configuration support for sslRejectUnauthorized: false #1247

rakesh561 opened the pull request:
PR to support postgres-configuration support for sslRejectUnauthorized: false #1247

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->

<!---
If you are fixing an existing issue, please include "fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New or Enhanced Feature

##### ADDITIONAL INFORMATION
<!---
Include additional information to help people understand the change here.
For bugs that don't have a linked bug report, a step-by-step reproduction
of the problem is helpful.
-->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```

```

rakesh561 committed:
PR to support postgres-configuration support for sslRejectUnauthorized: false #1247

PR to support postgres-configuration support for sslRejectUnauthorized: false #1247

rakesh561 committed:
Update credentials.py.j2

TheRealHaoLiu commented on roles/installer/tasks/database_configuration.yml:
```diff
@@ -85,21 +85,22 @@
 
 - name: Set PostgreSQL Configuration
   set_fact:
```
now there's 2 set_fact

TheRealHaoLiu commented on roles/installer/tasks/database_configuration.yml:
```diff
@@ -85,21 +85,22 @@
 
 - name: Set PostgreSQL Configuration
   set_fact:
-    pg_config: '{{ _generated_pg_config_resources["resources"] | default([]) | length | ternary(_generated_pg_config_resources, _pg_config) }}'
```
is this removal intended?

rakesh561 replied:
That was not intended.I updated the PR set_fact got added twice in the same task.

rakesh561 replied:
removed the extra set_Fact that was causing issue

rooftopcellist commented on roles/installer/tasks/database_configuration.yml:
```diff
@@ -85,21 +85,22 @@
 - name: Set PostgreSQL Configuration
   set_fact:
-    pg_config: '{{ _generated_pg_config_resources["resources"] | default([]) | length | ternary(_generated_pg_config_resources, _pg_config) }}'
-  no_log: "{{ no_log }}"
```
I think this removal is unintended. It results in invalid syntax:
* https://github.com/ansible/awx-operator/blob/6dbe2aa134d1a4ebd03a70bc584921eff54914d6/roles/installer/tasks/database_configuration.yml#L86-L89

rakesh561 replied:
That was not intended.I updated the PR set_fact got added twice in the same task.

rakesh561 committed:
Removed the extra set_fact line that got added accidentally

Removed the extra set_fact line that got added accidentally

rakesh561 commented:
Any suggestions/guidance in case of issues to fix them would be much appreciated would like to contribute to the open source.

fosterseth commented on roles/installer/tasks/database_configuration.yml:
```diff
@@ -85,21 +85,21 @@
     awx_postgres_port: "{{ pg_config['resources'][0]['data']['port'] | b64decode }}"
     awx_postgres_host: "{{ pg_config['resources'][0]['data']['host'] | b64decode }}"
     awx_postgres_sslmode: "{{ pg_config['resources'][0]['data']['sslmode'] |  default('prefer'|b64encode) | b64decode }}"
+    awx_postgres_sslunauthorized: "{{ pg_config['resources'][0]['data']['sslrejectUnauthorized'] |  default('false'|b64encode) | b64decode }}"
```
do you mind adding a note about this setting to the operator docs, somewhere here

https://github.com/ansible/awx-operator/blob/6513e93e7fdf117a1518392a0d838f267a67088d/README.md?plain=1#L536

you don't need to change the example spec above that line though, just a note about this param

rakesh561 replied:
Done @fosterseth also default value i have set it to false do you think it has to be set to true and then if some one uses external database without sslrejectUnauthorized they can pass the variable false.

rakesh561 committed:
Updated the Read me which includes information about sslrejectUnauthorized while using external postgres database

Updated the Read me which includes information about sslrejectUnauthorized while using external postgres database

dsavineau commented on README.md:
```diff
@@ -533,6 +533,8 @@ type: Opaque
 
 > It is possible to set a specific username, password, port, or database, but still have the database managed by the operator. In this case, when creating the postgres-configuration secret, the `type: managed` field should be added.
 
+**Note**. The vairable `sslrejectUnauthorized1 is valid for `external` databases only. The allowed values are: `false`, `true`.
```
```suggestion
**Note**. The variable `sslrejectUnauthorized` is valid for `external` databases only. The allowed values are: `false`, `true`.
```

dsavineau reviewed (commented):
Could you share where's the `sslrejectUnauthorized` option is coming from ?

I've looked at the libpq and psycopg documentation/code and never seen such option.
The closest I found was in a postgresql nodejs binding project.

Also if this option is only needed for external database then we need to set this option only if the database type is `unmanaged`

rakesh561 commented:
@dsavineau seems like sslrejectUnauthorized` is not available in the python postgres library i will close this request
